src/transform/src/lib.rs

```diff
@@ -819,6 +819,40 @@ impl Optimizer {
         }
     }
 
+    /// Builds a tiny optimizer, which is only suitable for optimizing the MIR plans that come from
+    /// constant INSERTs, e.g.
+    /// ```code
+    /// Return
+    ///   FlatMap wrap2(1, 2)
+    ///     Get l0
+    /// With
+    ///  cte l0 =
+    ///     Constant
+    ///       - ()
+    /// ```
+    /// which is planned for stuff like `INSERT INTO t VALUES (1,2);`.
+    ///
+    /// The important thing here is dealing with the `wrap` table function, which is always planned
+    /// for `VALUES`, and results in a Let binding in MIR. This is why we need to call
+    /// `NormalizeLets`, so that this Let binding is inlined. Then, `FoldConstants` is able to
+    /// evaluate the table function.
+    ///
+    /// The `InsertMultiRow` test is meant to make noise if this doesn't actually reach a constant
+    /// for a constant INSERT.
+    pub fn constant_insert_optimizer(_ctx: &mut TransformCtx) -> Self {
+        let transforms: Vec<Box<dyn Transform>> = vec![
+            Box::new(NormalizeLets::new(false)),
+            Box::new(canonicalization::ReduceScalars),
```

Comment:
I don't follow why `ReduceScalars` is here. It doesn't do relation constant folding, but the other two do.

Reply:
Just in case a user writes something like
```
insert into t values (1+1+1+1);
```

Reply:
(This is now included in `fold_constants_fixpoint()`.)